runtime/src/network/iouring.rs

```diff
@@ -69,9 +69,16 @@ impl Network {
     /// large enough, given the number of connections that will be maintained.
     /// Each ongoing send/recv to/from each connection will consume a slot in the io_uring.
     /// The io_uring `size` should be a multiple of the number of expected connections.
-    pub(crate) fn start(cfg: Config, registry: &mut Registry) -> Result<Self, crate::Error> {
+    pub(crate) fn start(mut cfg: Config, registry: &mut Registry) -> Result<Self, crate::Error> {
         // Create an io_uring instance to handle send operations.
         let (send_submitter, rx) = mpsc::channel(cfg.iouring_config.size as usize);
+
+        // Optimize performance by hinting the kernel that a single task will
+        // submit requests. This is safe because each iouring instance runs in a
+        // dedicated thread, which guarantees that the same thread that creates
+        // the ring is the only thread submitting work to it.
+        cfg.iouring_config.single_issuer = true;
```

Comment:
I pondered suggesting that we set this in `tokio` before calling this (as we provide the ring config): https://github.com/commonwarexyz/monorepo/blob/d0964ace3eea67c123b456901e04da867c0dc9e0/runtime/src/tokio/runtime.rs#L291-L298

But, that does require us to leak the abstraction here that we are using a single thread.

Reply:
On balance, I think overwriting to use `single_issuer` here is fine. Perhaps we could just add a comment on `run` noting that we do so?